presto-main/src/main/java/com/facebook/presto/server/StatementResource.java

```diff
@@ -164,13 +165,16 @@ public Response createQuery(
     {
         assertRequest(!isNullOrEmpty(statement), "SQL statement is empty");
 
-        Session session = createSessionForRequest(servletRequest, accessControl, sessionPropertyManager, queryIdGenerator.createNextQueryId());
-
-        ExchangeClient exchangeClient = exchangeClientSupplier.get(deltaMemoryInBytes -> { });
-        Query query = new Query(session, statement, queryManager, exchangeClient);
-        queries.put(query.getQueryId(), query);
-
-        return getQueryResults(query, Optional.empty(), uriInfo, new Duration(1, MILLISECONDS));
+        try {
+            Session session = createSessionForRequest(servletRequest, accessControl, sessionPropertyManager, queryIdGenerator.createNextQueryId());
+            ExchangeClient exchangeClient = exchangeClientSupplier.get(deltaMemoryInBytes -> { });
+            Query query = new Query(session, statement, queryManager, exchangeClient);
+            queries.put(query.getQueryId(), query);
+            return getQueryResults(query, Optional.empty(), uriInfo, new Duration(1, MILLISECONDS));
+        }
+        catch (PrestoException e) {
```

Comment:
The exception is thrown when something will go wrong while creating a session for the query. The only example I know about is when server is still initializing, and `InitializingSystemAccessControl::checkCanSetUser(principal, userName)` is used.
It looks like a proper behavior too me, but if you wan't it to be changed - please let me know what's expected here.

This also explains why I thought creating `PrestoExceptionSerialized` is better idea (as we throw it before Query is even created it seemed wrong to use `QueryError`). Having yours example though, I'm more than happy to drop `PrestoExceptionSerialized` in favor of `QueryError`.